torch/csrc/jit/tracer.cpp

```diff
@@ -203,13 +203,25 @@ Value* TracingState::getOutput(const IValue& iv, size_t i) {
      return it->second;
   } else if (iv.isTuple()) {
     auto tuple = iv.toTuple()->elements();
-    auto tuple_node = graph->createTuple(
-        fmap(tuple, [&](const IValue& ival) { return getOutput(ival, i); }));
+    TupleTypePtr tuple_type = iv.type()->cast<TupleType>();
+    Node* tuple_node = nullptr;
```

Comment:
No need for this if statement, the type can be passed unconditionally. createTuple will work if the type passed is not a named tuple as well.

Reply:
There's an assert that we either not passing tuple type or the tuple type is a named tuple https://github.com/pytorch/pytorch/blob/309b28ee3a4ba8f916b6a377fe0d3d11d87f2592/torch/csrc/jit/ir.cpp#L1448

I will remove that assert then.